### PR TITLE
Detection api update

### PR DESCRIPTION
**Pull request by JordanCheney**
@sklum I think this is ready for review. The purpose of this is to facilitate representations that use metadata as well as the image.

### CODE REVIEW

File: openbr/core/boost.cpp [hunk 1]
```diff
@@ -131,23 +131,17 @@ static CvMat* cvPreprocessIndexArray( const CvMat* idx_arr, int data_arr_size, b
...
 //------------------------------------- FeatureEvaluator ---------------------------------------
 
-void FeatureEvaluator::init(Representation *_representation, int _maxSampleCount, int channels)
+void FeatureEvaluator::init(Representation *_representation, int _maxSampleCount)
 {
     representation = _representation;
-
-    int dx, dy;
-    Size windowSize = representation->windowSize(&dx, &dy);
-    data.create((int)_maxSampleCount, (windowSize.width + dx) * (windowSize.height + dy), CV_32SC(channels));
```
Comment (sklum) [inline openbr/core/boost.cpp]: I need the ability to specify number of channels.

File: openbr/plugins/representation/random.cpp [hunk 2]
```diff
@@ -40,26 +40,14 @@ class RandomRepresentation : public Representation
...
+        return representation->preprocess(src);
     }
 
-    float evaluate(const Mat &image, int idx) const
+    float evaluate(const Template &src, int idx) const
     {
-        return representation->evaluate(image,features[idx]);
-    }
-
-    Mat evaluate(const Mat &image, const QList<int> &indices) const
```
Comment (sklum) [inline openbr/plugins/representation/random.cpp]: Why was this removed entirely?

File: openbr/plugins/representation/haar.cpp [hunk 3]
```diff
@@ -39,64 +39,64 @@ class HaarRepresentation : public Representation
 
     void init()
     {
-        if (features.isEmpty()) {
```
Comment (sklum) [inline openbr/plugins/representation/haar.cpp]: This is actually important.  Init is called twice in some circumstances so the number of features you get will be doubled.

File: openbr/core/boost.cpp (hunk 1 above)
Comment (JordanCheney) [inline openbr/core/boost.cpp]: Since the entire template is stored the whole matrix is available as is to the representation so you shouldn't have to specify it here anymore

File: openbr/plugins/representation/random.cpp (hunk 2 above)
Comment (JordanCheney) [inline openbr/plugins/representation/random.cpp]: No class called evaluate with a list of indices every class just did it with a single index so I thought it could be removed

File: openbr/plugins/representation/haar.cpp (hunk 3 above)
Comment (JordanCheney) [inline openbr/plugins/representation/haar.cpp]: Why is init called twice?

File: openbr/plugins/representation/random.cpp (hunk 2 above)
Comment (JordanCheney) [inline openbr/plugins/representation/random.cpp]: I'm changing my mind here. I'll leave both evaluates (single index and list of indices)

File: openbr/core/boost.cpp (hunk 1 above)
Comment (sklum) [inline openbr/core/boost.cpp]: I see.  Didn't notice that we're just creating the cls matrix now.

File: openbr/plugins/representation/haar.cpp (hunk 3 above)
Comment (sklum) [inline openbr/plugins/representation/haar.cpp]: Not sure.

Comment (JordanCheney) [inline openbr/plugins/representation/haar.cpp]: Ok I'll add this in for all representations then

**Comment by JordanCheney** [comment]
@sklum I think this has all of the changes

**Comment by JordanCheney** [comment]
Ok the file is gone

**Comment by sklum** [comment]
lgtm
